File: Encryption-Decryption/Caesar-Cipher.cpp
#include <iostream>
using namespace std;

string encrypt(string text,int key)
{
    string result = "";
    for(int i=0;i<text.length();i++)
    {
        //checking for the upper case letters
        if(isupper(text[i]))
        {
            result += char(int(text[i]+key-65)%26 + 65);
        }
        //checking for the lower case letters
        else
        {
            result += char(int(text[i]+key-97)%26 + 97);
        }
    }
    return result;
}

string decrypt(string text,int key)
{
    string result = "";
    for(int i=0;i<text.length();i++)
    {
        //checking for the upper case letters
        if(isupper(text[i]))
        {
            result += char(int(text[i]-key-65)%26 + 65);
        }
        //checking for the lower case letters 
        else
        {
            result += char(int(text[i]-key-97)%26 + 97);
        }
    }
    return result;
}

int main()
{
    string text;
    cout<<"Enter a plain text: ";
    cin>>text;
    
    int key;
    cout<<"Enter a key-shift: ";
    cin>>key;
    
    cout<<"Encrypted text: "<<text<<endl;
    cout<<"Key-shift: "<<key<<endl;
    cout<<"Decrypted text: "<<decrypt(text,key)<<endl;
}

File: CaesarCipher.py
from cgitb import text
from unittest import result


def encrypt(text, k):
    result = ""
    for i in range(len(text)):
        char = text[i]
        # Encrypting the uppercase letter
        if (char.isupper()):
            result += chr((ord(char) + k-65) % 26 + 65)
        # Encrypting the lowercase letter
        else:
            result += chr((ord(char) + k-97) % 26 + 97)
    return result


def decrypt(cipherText, k):
    result = ""
    for i in range(len(cipherText)):
        char = cipherText[i]
        if (char.isupper()):
            result += chr((ord(char)-k-65) % 26 + 65)
        else:
            result += chr((ord(char)-k-97) % 26 + 97)
    return result


text = "COMPUTER NETWORKING LAB"
s = 4
print("Text  : " + text)
print("Shift : " + str(s))
print("Cipher: " + encrypt(text, s))

cipherText = "GSQTYXIVrRIXASVOMRKrPEF"
k = 4
print("Plain Text: " + decrypt(cipherText, k))
